packages/expo-clipboard/src/web/WebExpoClipboard.ts

```diff
@@ -0,0 +1,136 @@
+      } catch (e) {
+        return Promise.reject(new Error('Unable to retrieve item from clipboard.'));
+      }
+    }
+    return text;
+  },
+  setString(text: string): boolean {
+    let success = false;
+    const textField = document.createElement('textarea');
+    textField.textContent = text;
+    document.body.appendChild(textField);
+    textField.select();
+    try {
+      document.execCommand('copy');
+      success = true;
+    } catch (e) {}
+    document.body.removeChild(textField);
+    return success;
+  },
+  async setStringAsync(text: string, _options: SetStringOptions): Promise<boolean> {
+    return this.setString(text);
+  },
+  async hasStringAsync(): Promise<boolean> {
+    return this.getStringAsync({}).then((text) => text.length > 0);
+  },
+  async getImageAsync(_options: GetImageOptions): Promise<ClipboardImage | null> {
+    if (!navigator.clipboard) {
+      throw new ClipboardUnavailableException();
+    }
+
+    try {
+      const clipboardItems = await navigator.clipboard.read();
+      const blob = await findImageInClipboardAsync(clipboardItems);
+      if (!blob) {
+        return null;
+      }
+
+      const [data, [width, height]] = await Promise.all([
+        blobToBase64Async(blob),
+        getImageSizeFromBlobAsync(blob),
+      ]);
+
+      return {
+        data,
+        size: { width, height },
+      };
+    } catch (e) {
+      // it might fail, because user denied permission
+      if (e.name === 'NotAllowedError' || (await isClipboardPermissionDeniedAsync())) {
+        throw new NoPermissionException();
+      }
+      throw new PasteFailureException(e.message);
+    }
+  },
+  async setImageAsync(base64image: string): Promise<void> {
+    if (!navigator.clipboard) {
+      throw new ClipboardUnavailableException();
+    }
+
+    try {
+      // we set it always to `image/png` because it's the only format supported by the clipboard
+      // but it seems to work even when provided jpeg data
+      const blob = base64toBlob(base64image, 'image/png');
+      await navigator.clipboard.write([
+        // I cannot use `@ts-expect-error` here because some environments consider this correct:
+        // expo-module build - OK,
+        // et gdad - error
+        // Fixed in TS >4.4.3: https://github.com/microsoft/TypeScript/issues/46116#issuecomment-932443415
+        // @ts-ignore Some tools seem to use TS <= 4.4.3
+        new ClipboardItem({
+          [blob.type]: blob,
+        } as Record<string, ClipboardItemDataType>),
+      ]);
+    } catch (err: any) {
+      throw new CopyFailureException(err.message);
+    }
+  },
+  async hasImageAsync(): Promise<boolean> {
+    if (!navigator.clipboard) {
+      throw new ClipboardUnavailableException();
+    }
+
+    try {
+      const clipboardItems = await navigator.clipboard.read();
+      return clipboardItems
+        .flatMap((item) => item.types)
+        .some((type) => type === 'image/png' || type === 'image/jpeg');
```

Comment:
Can you share this logic with `findImageInClipboardAsync`?
I imagine you can have:
```ts
async function readImagesFromClipboard() {
  const clipboardItems = await navigator.clipboard.read();
  return return clipboardItems
    .filter(item => item.types.some(type => ['image/png', 'image/jpeg'].contains(type))
}

const images = readImagesFromClipboard();

// 1. hasImageAsync
return hasImagesAsync.length > 0

// 2. getImageAsync
async function readImageBlobFromClipboardItem(item: ClipboardItem?) {
  // first look for png
    if (item?.types.some((type) => type === 'image/png')) {
      return item.getType('image/png');
    }

    // alternatively, an image might be a jpeg
    // NOTE: Currently, this is not supported by browsers yet. They only support PNG now
    if (item?.types.some((type) => type === 'image/jpeg')) {
      return clipboardItem.getType('image/jpeg');
    }
    
    return null
}
const blob = await readImageBlobFromClipboardItem(images[0])